### ISSUE TITLE
TypeError when accessing search entries if _schema.attribute_types is not set.

### ISSUE THREAD

**Issue by pmariani** [cannatag/ldap3#252]
Hi

I have the following test code:

```python
import ldap3

LDAP_URL = 'some-url'
PASSWORD = 'secret'
USERNAME = 'uid=pierre,ou=people,dc=initech,dc=com'

server = ldap3.Server(LDAP_URL, get_info=ldap3.ALL)
print(server.schema)
connection = ldap3.Connection(server, USERNAME, PASSWORD, auto_bind=True)
connection.search(USERNAME, '(uid=*)', attributes=ldap3.ALL_ATTRIBUTES, search_scope=ldap3.BASE)
ent = connection.entries[0]
try:
    print(ent.entry_get_attributes_dict())
except ldap3.core.exceptions.LDAPCursorError:
    print(ent.entry_attributes_as_dict)
```
Using 2.1.0, this code fails with the following stacktrace:

```python
Traceback (most recent call last):
  File "test-ldap.py", line 11, in <module>
    ent = connection.entries[0]
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/core/connection.py", line 1389, in entries
    self._entries = self._get_entries(self.response)
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/core/connection.py", line 1414, in _get_entries
    object_def += list(attr_set)  # converts the set in a list to be added to the object definition
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/abstract/objectDef.py", line 131, in __iadd__
    self.add_attribute(other)
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/abstract/objectDef.py", line 188, in add_attribute
    self.add_attribute(element)
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/abstract/objectDef.py", line 180, in add_attribute
    self.add_from_schema(definition)
  File "/Users/pierre/emacsvenv/lib/python3.5/site-packages/ldap3/abstract/objectDef.py", line 169, in add_from_schema
    if attribute_name in self._schema.attribute_types:
TypeError: argument of type 'NoneType' is not iterable

```
The LDAP server happens to not return a schema, and self._schema.attributes_type is None at the line specified.

If I apply the following patch, I get the result I expect.

```patch
diff --git a/ldap3/abstract/objectDef.py b/ldap3/abstract/objectDef.py
index 96e0f24..c32a24c 100644
--- a/ldap3/abstract/objectDef.py
+++ b/ldap3/abstract/objectDef.py
@@ -165,7 +165,7 @@ class ObjectDef(object):
         attr_def = AttrDef(attribute_name)
         attr_def.validate = find_attribute_validator(self._schema, attribute_name, self._custom_validator)
         attr_def.mandatory = mandatory  # in schema mandatory is specified in the object class, not in the attribute class
-        if self._schema:
+        if self._schema and self._schema.attribute_types:
             if attribute_name in self._schema.attribute_types:
                 attr_def.single_value = self._schema.attribute_types[attribute_name].single_value
                 attr_def.oid_info = self._schema.attribute_types[attribute_name]

```

**Comment by cannatag** [comment]
Updated in dev. Can you check it?

**Comment by pmariani** [comment]
Works, thanks!

**Comment by cannatag** [comment]
will be released in 2.1.1

**Comment by cannatag** [comment]
Released in 2.1.1
